File: fe/mobile/android/app/src/main/java/com/capstone/bridge/PermissionBridgePackage.java
package com.capstone.bridge;

import com.capstone.infrastructure.security.PermissionManager;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactPackage;
import com.facebook.react.uimanager.ViewManager;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class PermissionBridgePackage implements ReactPackage {
    private final PermissionManager permissionManager;

    public PermissionBridgePackage(PermissionManager permissionManager) {
        this.permissionManager = permissionManager;
    }

    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        List<NativeModule> modules = new ArrayList<>();
        modules.add(new PermissionBridgeModule(reactContext, permissionManager));
        return modules;
    }

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }
}

File: fe/mobile/android/app/src/main/java/com/capstone/bridge/PermissionBridgePackage.kt
package com.capstone.bridge

import android.view.View
import com.capstone.infrastructure.security.PermissionManager
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ReactShadowNode
import com.facebook.react.uimanager.ViewManager

class PermissionBridgePackage(
    private val permissionManager: PermissionManager
) : ReactPackage {
    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
        return listOf(PermissionBridgeModule(reactContext, permissionManager))
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<View, ReactShadowNode<*>>> {
        return emptyList()
    }
}
